app/models/course/lesson_plan/item.rb

```diff
@@ -152,25 +143,43 @@ def find_or_create_personal_time_for(course_user)
     personal_time
   end
 
-  # Finds the lesson plan items which are starting within the next day for a given course.
+  def self.upcoming_for_exists?(course)
```

Comment:
I am thinking if we can simplify or remove this check as it's quite similar to the logic below for retrieving the opening items for each course user. But let's merge this PR first so that you can proceed.

Reply:
Also maybe this should be an instance method of `Course` model instead?